smrf/data/gridded_input.py

```diff
@@ -0,0 +1,37 @@
+import logging
+
+
+class GriddedInput:
+    TYPE = 'gridded'
+
+    def __init__(self,
+                 start_date, end_date,
+                 bbox=None, topo=None, config=None):
+
+        self.start_date = start_date
+        self.end_date = end_date
+        self.time_zone = start_date.tzinfo
+
+        self.__set_attribute('topo', topo)
+        self.__set_attribute('bbox', bbox)
+        self.__set_attribute('config', config)
+
+        self._logger = logging.getLogger(self.__class__.__name__)
+
+    @property
+    def bbox(self):
+        return self._bbox
+
+    @property
+    def config(self):
+        return self._config
+
+    @property
+    def topo(self):
+        return self._topo
+
+    def __set_attribute(self, attribute, argument):
```

Comment:
Wouldn't this be easier to just remove the `None` default argument? Is this trying to make less code instead of specifying setters for each one?

Reply:
I went the conservative route by not changing the method signature. I agree that just removing 'None' would be a lot easier. Go for the bigger change then?

Reply:
Thinking about this a bit more, an advantage of this handling is that the error message will be more informative by giving the exact parameter that is missing. Removing the `None` would simply result in a "wrong number of arguments" error. Food for thought ...

Reply:
Agree that it's more verbose the way you have it. I'm confused as to why the original was set to None to begin with then checked to make sure it wasn't None... I'll chock it up to learning python!

I'll leave it up to you, it seems that you have to have those parameters passed for the gridded data.

Reply:
Let me go through the code one more time and see if the 'None` can be dropped. Do I hear that there are no other consumers of the class other than internal SMRF?

Reply:
It should just be internal to SMRF, I don't know anything outside that uses it. Even AWSM wouldn't use it.

Reply:
What do you think of this approach?

Reply:
Looks good, gives more verbose errors when they happen.

Reply:
I am ready to merge then ....

Reply:
Trying to review it now, we're still having VPN issues...